superset-frontend/src/embedded/index.tsx

```diff
@@ -154,15 +157,39 @@ function setupGuestClient(guestToken: string) {
 }
 
 function validateMessageEvent(event: MessageEvent) {
-  // if (!ALLOW_ORIGINS.includes(event.origin)) {
-  //   throw new Error('Message origin is not in the allowed list');
-  // }
-
   if (typeof event.data !== 'object' || event.data.type !== MESSAGE_TYPE) {
     throw new Error(`Message type does not match type used for embedded comms`);
   }
 }
 
+function validateAllowedDomains() {
+  const uuid = window.location.pathname.replace('/embedded/', '');
+  const getDashaboardConfig = makeApi<void, { result: EmbeddedDashboard }>({
+    method: 'GET',
+    endpoint: `/api/v1/embedded_dashboard/${uuid}`,
+  });
+  return getDashaboardConfig().then(
+    ({ result }) => {
+      const parentOrigin = new URL(document.referrer).origin;
+      if (
+        result.allowed_domains.length > 0 &&
+        !result.allowed_domains.includes(parentOrigin)
+      ) {
+        showFailureMessage(
+          `iframe parent ${parentOrigin} is not in the list of allowed domains`,
+        );
+      }
+    },
+    err => {
```

Comment:
Looks like you copied this from the `getMe` call in the `start` function, but that endpoint is specifically being used to check guest token validity. This error handler is inappropriate outside that case.

Reply:
oops i will update the message